File: event-service/src/main/java/com/ehub/event/service/TeamService.java
package com.ehub.event.service;

import java.util.List;
import java.util.Map;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import com.ehub.event.dto.TeamCreateRequest;
import com.ehub.event.dto.TeamInviteRequest;
import com.ehub.event.dto.TeamResponse;
import com.ehub.event.dto.TeamSubmissionRequest;
import com.ehub.event.facade.TeamFacade;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class TeamService {

    private final TeamFacade teamFacade;

    public void createTeam(String eventId, TeamCreateRequest request, String userId) {
        teamFacade.createTeam(eventId, request, userId);
    }

    public List<TeamResponse> getTeamsByEvent(String eventId, String name) {
        return teamFacade.getTeamsByEvent(eventId, name);
    }

    public Page<TeamResponse> getTeamsByEvent(String eventId, String name, Pageable pageable) {
        return teamFacade.getTeamsByEvent(eventId, name, pageable);
    }

    public TeamResponse getTeamByShortCode(String shortCode) {
        return teamFacade.getTeamByShortCode(shortCode);
    }

    public void inviteMember(String teamId, TeamInviteRequest request, String requesterId) {
        teamFacade.inviteMember(teamId, request, requesterId);
    }

    public void requestToJoin(String teamId, TeamInviteRequest request) {
        teamFacade.requestToJoin(teamId, request);
    }

    public void respondToInvite(String teamId, String userId, boolean accept) {
        teamFacade.respondToInvite(teamId, userId, accept);
    }

    public void respondToJoinRequest(String teamId, String leaderId, String requestingUserId, boolean accept) {
        teamFacade.respondToJoinRequest(teamId, leaderId, requestingUserId, accept);
    }

    public void dismantleTeam(String teamId, String leaderId) {
        teamFacade.dismantleTeam(teamId, leaderId);
    }

    public void transferLeadership(String teamId, String currentLeaderId, String newLeaderId) {
        teamFacade.transferLeadership(teamId, currentLeaderId, newLeaderId);
    }

    public void leaveTeam(String teamId, String userId) {
        teamFacade.leaveTeam(teamId, userId);
    }

    public void updateProblemStatement(String teamId, String leaderId, String problemId) {
        teamFacade.updateProblemStatement(teamId, leaderId, problemId);
    }

    public void submitProject(String teamId, String userId, TeamSubmissionRequest request) {
        teamFacade.submitProject(teamId, userId, request);
    }

    public Map<String, Object> getTeamForEvaluation(String teamId) {
        return teamFacade.getTeamForEvaluation(teamId);
    }

    public List<Map<String, Object>> getEventEvaluationContext(String eventId) {
        return teamFacade.getEventEvaluationContext(eventId);
    }

    public void updateScore(String teamId, Double score, String aiSummary, String requesterId) {
        teamFacade.updateScore(teamId, score, aiSummary, requesterId);
    }

    public void updateManualReview(String teamId, Double manualScore, String organizerNotes, String requesterId) {
        teamFacade.updateManualReview(teamId, manualScore, organizerNotes, requesterId);
    }

    public void updateSkillsNeeded(String teamId, List<String> skills, String requesterId) {
        teamFacade.updateSkillsNeeded(teamId, skills, requesterId);
    }
}

File: client-service/src/services/teamService.js
import api from './api';

/**
 * Team service — full lifecycle management for hackathon teams.
 *
 * Covers: creation, membership (join requests, invitations, responses, transfers, leave, dismantle),
 * project submission, problem statement selection, skills/matchmaking, and judging/manual review.
 *
 * Note: several methods accept a `_leaderId` / `_userId` param that is currently unused by the API
 * (auth is enforced server-side via JWT) — kept for call-site clarity.
 */
const teamService = {
  getTeamsByEvent:      (eventId, name, params)    => api.get(`/events/teams/${eventId}`, { params: { ...(name ? { name } : {}), ...params } }).then(r => r?.content ?? r),
  getTeamByShortCode:   (shortCode)               => api.get(`/events/teams/code/${shortCode}`),
  createTeam:           (eventId, data)            => api.post(`/events/teams/${eventId}`, data),
  inviteMember:         (teamId, _leaderId, data)  => api.post(`/events/teams/${teamId}/invite`, data),
  requestToJoin:        (teamId, data)             => api.post(`/events/teams/${teamId}/request`, data),
  respondToInvite:      (teamId, _userId, accept)  => api.patch(`/events/teams/${teamId}/respond`, null, { params: { accept } }),
  dismantleTeam:        (teamId, _leaderId)        => api.delete(`/events/teams/${teamId}`),
  transferLeadership:   (teamId, _cLId, newLeaderId) => api.patch(`/events/teams/${teamId}/transfer`, null, { params: { newLeaderId } }),
  leaveTeam:            (teamId, _userId)          => api.delete(`/events/teams/${teamId}/leave`),
  selectProblemStatement:(teamId, _lId, problemId) => api.patch(`/events/teams/${teamId}/problem-statement`, null, { params: problemId ? { problemId } : {} }),
  submitProject:        (teamId, _userId, data)    => api.post(`/events/teams/${teamId}/submit`, data),
  updateSkillsNeeded:   (teamId, skills)           => api.patch(`/events/teams/${teamId}/skills-needed`, { skills }),
  suggestMembers:       (teamId, params)           => api.get(`/events/matchmaking/suggest-members/${teamId}`, { params }),
  // Judging / results (domain: team scoring)
  updateManualReview:   (teamId, data)             => api.patch(`/events/teams/${teamId}/manual-review`, data),
  finalizeResults:      (eventId)                  => api.patch(`/events/${eventId}/finalize-results`),
};

export default teamService;
